perun-rpc/src/main/java/cz/metacentrum/perun/rpc/Api.java

```diff
@@ -834,6 +842,71 @@ private Deserializer selectDeserializer(String format, HttpServletRequest req) t
 		}
 	}
 
+	private record OidcSubIssuer(String sub, String issuer) {}
+
+	private static OidcSubIssuer initOidcAuth(HttpServletRequest req, Map<String, String> additionalInformations) {
+		var accessToken = req.getHeader(OIDC_ACCESS_TOKEN);
+		var restTemplate = new RestTemplate();
+		var issuer = req.getHeader(OIDC_CLAIM_ISS);
+		var config = restTemplate.getForObject(issuer + "/.well-known/openid-configuration", JsonNode.class);
+		var headers = new HttpHeaders();
+		headers.setBearerAuth(accessToken);
+		HttpEntity<Void> entity = new HttpEntity<>(headers);
+		var userInfoResponse = restTemplate.exchange(config.path("userinfo_endpoint").textValue(), HttpMethod.GET, entity, JsonNode.class);
+		var userInfo = userInfoResponse.getBody();
+		if(isNotEmpty(userInfo.path("error").asText())) {
+			log.error("Call to user info endpoint failed, the error is: {}", userInfo);
+			throw new InternalErrorException("Call to user info endpoint failed, the error is" + userInfo);
+		}
+		log.debug("user info retreated: {}", userInfo);
+		var name = userInfo.path("name").asText();
+		if(isEmpty(name)) {
+			var firstName = userInfo.path("given_name").asText();
+			var familyName = userInfo.path("family_name").asText();
+			if(isNotEmpty(firstName) && isNotEmpty(familyName)) {
+				name =  firstName + " " + familyName;
+			}
+		}
+		var email = userInfo.path("email").asText();
+		if(isNotEmpty(email)) {
+			additionalInformations.put("mail", email);
+		}
+		if(isNotEmpty(name)) {
+			additionalInformations.put("displayName", name);
+		}
+		var idpNames = userInfo
+			.path("target_backend")
+			.path("display_name")
+			.elements();
+		log.debug("found idp names: {}", idpNames);
+		if(idpNames.hasNext()) {
+			var idpName = idpNames.next().path("text").asText();
+			log.debug("found idp name: {}", idpName);
+			additionalInformations.put("sourceIdPName", idpName);
+		}
+		var originalIssuer = userInfo.path("target_issuer").asText();
+		if(isEmpty(originalIssuer)) {
+			log.error("issuer was empty or null: {}", originalIssuer);
+			throw new InternalErrorException("issuer was null: " + originalIssuer);
+		}
+		var login = userInfo.path("eduperson_unique_id").asText();
+		if(isEmpty(login)) {
+			login = userInfo.path("eduperson_principal_name").asText();
+			if(isEmpty(login)) {
+				login = userInfo.path("saml2_nameid_persistent").asText();
+				if(isEmpty(login)) {
+					var edupersonTargetedId = userInfo.path("eduperson_targeted_id").get(0);
+					login = (edupersonTargetedId == null) ? "" : edupersonTargetedId.asText();
+					if(isEmpty(login)) {
+						var voperson_external_id = userInfo.path("voperson_external_id").get(0);
+						login = (voperson_external_id == null) ? "" : voperson_external_id.asText();
+					}
+				}
+			}
+		}
```

Comment:
retrieving login (=original sub, = ExtsourceLogin) - unique 
all this properies proxy usually doesnt send, proxy must be modified